frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -145,44 +145,67 @@ export class VarianceHomePage extends Component {
     this.props.fetchVarianceDocumentCategoryOptions();
   }
 
+  componentWillReceiveProps(nextProps) {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      this.renderDataFromURL(nextProps);
+    }
+  }
+
   componentWillUnmount() {
     this.handleVarianceSearchDebounced.cancel();
     this.setState({
       params: {},
     });
   }
 
-  renderDataFromURL = (params) => {
+  renderDataFromURL = (nextProps) => {
+    const params = nextProps ? nextProps.location.search : this.props.location.search;
     const { region, compliance_code, major, search, ...remainingParams } = queryString.parse(
```

Comment:
the params names are confusing here - `nextProps` should only be used in `componentsWillReceiveProps`, as the data passed into this function is only params not all of the available props which this implies. 

You do a check for params in `componentWillReceiveProps`, would be better to pass them in at that point since `params` already  equal `this.props.location.search`

Then the function will already accept the proper params and this check will no longer be needed. 
`renderDataFromURL = (params) `

Reply:
done